### DISCUSSION TITLE
Does a `isRoute()` composable exist ... ?

### DISCUSSION THREAD

**Discussion by michealroberts** [vuejs/router#1820] [Help and Questions] [answered]
Just a quick query, I've scanned the documentation for the potential of something like:

```ts
isRoute("hello-world") && router.push("hello-world")
```

Where we can lightly check if a route exists without the error when one isn't defined before routing ... does such a composable / utility exist? Maybe by a different name?

**Comment by posva** [comment, marked as answer]
It doesn't exist but you can create your own by scanning the array of routes `router.getRoutes()` which is already flattened or with `router.resolve()`.

FYI that wouldn't be a composable, it just be a function.

**Reply by michealroberts** [reply]
@posva I think it would / could be a composable in the sense that the route param could be reactive ...
